src/hooks/useInitializeApp.ts

```diff
@@ -1,6 +1,42 @@
 import { useFonts } from 'expo-font';
+import { useEffect, useState } from 'react';
+import { registerDeviceToken } from '@/api/notification';
+import { useAuthStore } from '@/stores/authStore';
+import { getFCMToken, onTokenRefresh, requestUserPermission } from '@/utils/firebase';
 
 export default function useInitializeApp() {
+  const [loaded, setLoaded] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const { setDeviceToken } = useAuthStore();
+
+  useEffect(() => {
+    async function initialize() {
+      try {
+        // FCM 권한 요청
+        await requestUserPermission();
+
+        // FCM 토큰 가져오기 및 서버에 등록
+        const token = await getFCMToken();
+        if (token) {
+          setDeviceToken(token);
+          await registerDeviceToken(token);
+        }
+
+        // 토큰 새로고침 리스너 등록
+        onTokenRefresh(async (newToken) => {
+          setDeviceToken(newToken);
+          await registerDeviceToken(newToken);
+        });
+
+        setLoaded(true);
+      } catch (e) {
+        setError(e as Error);
```

Comment:
단언 보다는 타입 지정해주는게 좋을 것 같습니다.
```suggestion
      } catch (e: Error) {
        setError(e);
```